XamarinCommunityToolkit/Helpers/BindablePropertyBuilder.cs

```diff
@@ -0,0 +1,177 @@
+		{
+			bpParameters.BindingPropertyChangedDelegate = propertyChangedDelegate;
+			return this;
+		}
+
+		/// <summary>
+		/// Sets the delegate that runs before a bound value is changed.
+		/// </summary>
+		/// <param name="propertyChangingDelegate">The delegate that runs before a bound value is changed.</param>
+		/// <returns>The <see cref="BindablePropertyBuilder"/> to support the fluent syntax</returns>
+		public BindablePropertyBuilder SetPropertyChangingDelegate(BindableProperty.BindingPropertyChangingDelegate propertyChangingDelegate)
+		{
+			bpParameters.BindingPropertyChangingDelegate = propertyChangingDelegate;
+			return this;
+		}
+
+		/// <summary>
+		/// Sets the delegate used to coerce the range of a value.
+		/// </summary>
+		/// <param name="coerceValueDelegate">A delegate used to coerce the range of a value.</param>
+		/// <returns>The <see cref="BindablePropertyBuilder"/> to support the fluent syntax</returns>
+		public BindablePropertyBuilder SetCoerceValueDelegate(BindableProperty.CoerceValueDelegate coerceValueDelegate)
+		{
+			bpParameters.CoerceValueDelegate = coerceValueDelegate;
+			return this;
+		}
+
+		/// <summary>
+		/// Sets the delegate used to initialize the default value for reference types.
+		/// </summary>
+		/// <param name="createDefaultValueDelegate">The delegate used to initialize the default value for reference types.</param>
+		/// <returns>The <see cref="BindablePropertyBuilder"/> to support the fluent syntax</returns>
+		public BindablePropertyBuilder SetCreateDefaultValueDelegate(BindableProperty.CreateDefaultValueDelegate createDefaultValueDelegate)
+		{
+			bpParameters.CreateDefaultValueDelegate = createDefaultValueDelegate;
+			return this;
+		}
+
+		/// <summary>
+		/// Instantiates the <see cref="BindableProperty"/> based on the values that have been set.
+		/// </summary>
+		/// <returns>An instance of <see cref="BindableProperty"/> with the values that have been set</returns>
+		/// <exception cref="ArgumentNullException">
+		/// If any of the following are null:
+		/// <see cref="BindableProperty.PropertyName"/>
+		/// <see cref="BindableProperty.ReturnType"/>
+		/// <see cref="BindableProperty.DeclaringType"/>
+		/// </exception>
+		/// <exception cref="ArgumentException">
+		/// If the type of the <see cref="BindableProperty.DefaultValue"/>
+		/// and <see cref="BindableProperty.ReturnType"/> don't match.
+		/// </exception>
+		public BindableProperty Build()
+		{
+			CheckParameter(!string.IsNullOrEmpty(bpParameters.PropertyName), nameof(bpParameters.PropertyName));
+			CheckParameter(bpParameters.ReturnType != default, nameof(bpParameters.ReturnType));
+			CheckParameter(bpParameters.DeclaringType != default, nameof(bpParameters.DeclaringType));
+			if (bpParameters.ReturnType != bpParameters.DefaultValue?.GetType())
+				throw new ArgumentException($"The return type is {bpParameters.ReturnType.Name}, but the default value is of type {bpParameters.DefaultValue.GetType().Name}");
```

Comment:
This check is wrong.

First of all ```bpParameters.DefaultValue.GetType()``` can cause null reference exception before raising argument exception.

But DefaultValue can be null, it means that it's impossible to get its type.
Or even if somebody decides to use DefaultValueDelegate instead.

I see two possible approaches for handling this case.

1) Just removing these two lines
2) Ori if we want to check the type before creating, we will have to make BindablePropertyBuilder generic type

```new BindablePropertyBuilder<TDeclaring, TReturn>()```